src/pageManager.js

```diff
@@ -706,14 +703,30 @@ export async function initialize() {
         }
     }
 
-    // Register the pageManager content script for all URLs permitted by the extension manifest
-    browser.contentScripts.register({
-        matches: permissions.getManifestOriginMatchPatterns(),
-        js: [{
-            file: pageManagerContentScript
-        }],
-        runAt: "document_start"
-    });
+    // Register the pageManager content script for all URLs permitted by the extension manifest.
+    // Only Firefox supports dynamic content script loading at this time.
+    try {
+        const browserInfo = await browser.runtime.getBrowserInfo();
```

Comment:
This API appears to be [Firefox-only](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getBrowserInfo#browser_compatibility), I tried to be defensive here in case that changes in the future.

This should probably be made more abstract so it can be reused elsewhere. We know that Chrome will ship dynamic content script registration eventually, but I suspect the current API won't be polyfilled. It's more likely that the new API will be the one standardized around.